tlsutil/config.go

```diff
@@ -375,20 +375,17 @@ func (c *Configurator) check(config Config, pool *x509.CertPool, cert *tls.Certi
 
 	// Ensure we have a CA and cert if VerifyIncoming is set
 	if config.anyVerifyIncoming() {
-		autoEncryptMsg := " AutoEncrypt only secures the connection between client and server and doesn't affect incoming connections on the client."
 		if pool == nil {
-			errMsg := "VerifyIncoming set, and no CA certificate provided!"
-			if config.AutoEncryptTLS {
-				errMsg += autoEncryptMsg
-			}
-			return fmt.Errorf(errMsg)
+			// both auto-config and auto-encrypt while being able to provide the agents certificate and some ca certs require
+			// verifying the connection to the server. Therefore its an error if we ever want to verify incoming certificates
+			// but have no CA certs
+			return fmt.Errorf("VerifyIncoming set but no CA certificates were provided")
 		}
-		if cert == nil {
-			errMsg := "VerifyIncoming set, and no Cert/Key pair provided!"
-			if config.AutoEncryptTLS {
-				errMsg += autoEncryptMsg
-			}
-			return fmt.Errorf(errMsg)
+
+		// We will use the auto_encrypt/auto_config cert for TLS in the incoming APIs when available. Therefore the check
+		// here will ensure that either we enabled one of those two features or a certificate and key were provided manually
+		if cert == nil && !config.AutoEncryptTLS {
+			return fmt.Errorf("VerifyIncoming set but neither a Cert/Key pair was provided nor was auto_encrypt/auto_config enabled")
```

Comment:
If anyone else can think of a better way of saying this I would love to hear it. The crux is that you either must provide a certificate and key in the config or have set `auto_encrypt.tls=true` or have set `auto_config.enabled=true`

Reply:
Maybe just flip it around to remove the negatives?

"VerifyIncoming requires either a Cert and Key pair in the configuration file, or auto_encrypt/auto_config be enabled".

Reply:
👍